## ##############################################################################
## ChIP-seq case studies
## ##############################################################################

df_h3k4me3_promoters <- data.frame(
  Title = "h3k4me3-promoters-benchmark",
  Description = "Differential peak calling in human H3K4me3 ChIP-seq data from ENCODE using predefined regions with mean coverage as covariate",
  Series = "ChIP-seq",
  M = 118171,
  Genome = "hg18",
  SourceType = "BAM",
  SourceUrl = paste("http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me3StdAlnRep1.bam",
                    "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k04me3StdAlnRep2V2.bam",
                    "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me3StdAlnRep1.bam",
                    "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me3StdAlnRep2.bam",
                    collapse = ","),
  SourceVersion = paste("ENCODE Jan 2011 Freeze", "ENCODE Mar 2012 Freeze",
                        "ENCODE Jan 2011 Freeze", "ENCODE Jan 2011 Freeze",
                        collapse = ","),
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = FALSE,
  DataProvider = "ENCODE",
  TestStatistic = "DESeq2",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Mean coverage",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_h3k4me3_csaw <- data.frame(
  Title = "h3k4me3-csaw-benchmark",
  Description = "Differential peak calling in human H3K4me3 ChIP-seq data from ENCODE using csaw with region width as covariate",
  Series = "ChIP-seq",
  M = 118171,
  Genome = "hg18",
  SourceType = "BAM",
  SourceUrl = paste("http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me3StdAlnRep1.bam",
                    "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k04me3StdAlnRep2V2.bam",
                    "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me3StdAlnRep1.bam",
                    "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me3StdAlnRep2.bam",
                    collapse = ","),
  SourceVersion = paste("ENCODE Jan 2011 Freeze", "ENCODE Mar 2012 Freeze",
                        "ENCODE Jan 2011 Freeze", "ENCODE Jan 2011 Freeze",
                        collapse = ","),
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = FALSE,
  DataProvider = "ENCODE",
  TestStatistic = "csaw",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Region width",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_h3k4me3_csaw_uninf <- data.frame(
  Title = "h3k4me3-csaw-uninf-benchmark",
  Description = "Differential peak calling in human H3K4me3 ChIP-seq data from ENCODE using csaw with uninformative covariate",
  Series = "ChIP-seq",
  M = 118171,
  Genome = "hg18",
  SourceType = "BAM",
  SourceUrl = paste("http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me3StdAlnRep1.bam",
                    "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k04me3StdAlnRep2V2.bam",
                    "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me3StdAlnRep1.bam",
                    "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me3StdAlnRep2.bam",
                    collapse = ","),
  SourceVersion = paste("ENCODE Jan 2011 Freeze", "ENCODE Mar 2012 Freeze",
                        "ENCODE Jan 2011 Freeze", "ENCODE Jan 2011 Freeze",
                        collapse = ","),
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = FALSE,
  DataProvider = "ENCODE",
  TestStatistic = "csaw",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_h3k4me3_csaw_cov <- data.frame(
  Title = "h3k4me3-csaw-cov-benchmark",
  Description = "Differential peak calling in human H3K4me3 ChIP-seq data from ENCODE using csaw with mean coverage as covariate",
  Series = "ChIP-seq",
  M = 118171,
  Genome = "hg18",
  SourceType = "BAM",
  SourceUrl = paste("http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me3StdAlnRep1.bam",
                    "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k04me3StdAlnRep2V2.bam",
                    "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me3StdAlnRep1.bam",
                    "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me3StdAlnRep2.bam",
                    collapse = ","),
  SourceVersion = paste("ENCODE Jan 2011 Freeze", "ENCODE Mar 2012 Freeze",
                        "ENCODE Jan 2011 Freeze", "ENCODE Jan 2011 Freeze",
                        collapse = ","),
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = FALSE,
  DataProvider = "ENCODE",
  TestStatistic = "csaw",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Mean coverage",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_cbp_csaw <- data.frame(
  Title = "cbp-csaw-benchmark",
  Description = "Differential peak calling in CREB-binding protein (CBP) knockout and wild-type mice using csaw with region width as covariate",
  Series = "ChIP-seq",
  M = 59596,
  Genome = "mm10",
  SourceType = "FASTQ",
  SourceUrl = paste(c("ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR114/007/SRR1145787/SRR1145787.fastq.gz",
                      "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR114/008/SRR1145788/SRR1145788.fastq.gz",
                      "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR114/009/SRR1145789/SRR1145789.fastq.gz",
                      "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR114/000/SRR1145790/SRR1145790.fastq.gz"),
                    collapse = ","),
  SourceVersion = NA_character_,
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA_character_,
  DataProvider = "Biochemistry, St Jude Children's Research Hospital",
  TestStatistic = "csaw",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Region width",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_cbp_csaw_uninf <- data.frame(
  Title = "cbp-csaw-uninf-benchmark",
  Description = "Differential peak calling in CREB-binding protein (CBP) knockout and wild-type mice using csaw with uninformative covariate",
  Series = "ChIP-seq",
  M = 59596,
  Genome = "mm10",
  SourceType = "FASTQ",
  SourceUrl = paste(c("ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR114/007/SRR1145787/SRR1145787.fastq.gz",
                      "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR114/008/SRR1145788/SRR1145788.fastq.gz",
                      "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR114/009/SRR1145789/SRR1145789.fastq.gz",
                      "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR114/000/SRR1145790/SRR1145790.fastq.gz"),
                    collapse = ","),
  SourceVersion = NA_character_,
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA_character_,
  DataProvider = "Biochemistry, St Jude Children's Research Hospital",
  TestStatistic = "csaw",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_cbp_csaw_cov <- data.frame(
  Title = "cbp-csaw-cov-benchmark",
  Description = "Differential peak calling in CREB-binding protein (CBP) knockout and wild-type mice using csaw with mean coverage as covariate",
  Series = "ChIP-seq",
  M = 59596,
  Genome = "mm10",
  SourceType = "FASTQ",
  SourceUrl = paste(c("ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR114/007/SRR1145787/SRR1145787.fastq.gz",
                      "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR114/008/SRR1145788/SRR1145788.fastq.gz",
                      "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR114/009/SRR1145789/SRR1145789.fastq.gz",
                      "ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR114/000/SRR1145790/SRR1145790.fastq.gz"),
                    collapse = ","),
  SourceVersion = NA_character_,
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA_character_,
  DataProvider = "Biochemistry, St Jude Children's Research Hospital",
  TestStatistic = "csaw",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Mean coverage",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)


## ##############################################################################
## Gene Set Analysis case studies
## ##############################################################################

df_gsa_mouse <- data.frame(
  Title = "mouse-benchmark",
  Description = "Gene Set Enrichment Analysis of Mouse Data Set",
  Series = "GSA",
  M = 6396,
  Genome = "GRCm38",
  SourceType = "FASTQ",
  SourceUrl = "https://doi.org/10.5281/zenodo.1475409",
  SourceVersion = "md5:c25141558ff5313a87a4ec7539279b84",
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA_character_,
  DataProvider = "ArrayExpress",
  TestStatistic = "Enrichment Score",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Gene Set Size",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_gsa_mouse_uninf <- data.frame(
  Title = "mouse-uninf-benchmark",
  Description = "Gene Set Enrichment Analysis of Mouse Data Set",
  Series = "GSA",
  M = 6396,
  Genome = "GRCm38",
  SourceType = "FASTQ",
  SourceUrl = "https://doi.org/10.5281/zenodo.1475409",
  SourceVersion = "md5:c25141558ff5313a87a4ec7539279b84",
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA_character_,
  DataProvider = "ArrayExpress",
  TestStatistic = "Enrichment Score",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_gsa_human <- data.frame(
  Title = "human-benchmark",
  Description = "Gene Set Enrichment Analysis of Human Data Set",
  Series = "GSA",
  M = 6783,
  Genome = "GRCh38",
  SourceType = "FASTQ",
  SourceUrl = "https://doi.org/10.5281/zenodo.1475409",
  SourceVersion = "md5:28508f1b03d6cfef75a7b537b7f186b6",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA_character_,
  DataProvider = "Genotype-Tissue Expression Project (GETx)",
  TestStatistic = "Enrichment Score",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Gene Set Size",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_gsa_human_uninf <- data.frame(
  Title = "human-uninf-benchmark",
  Description = "Gene Set Enrichment Analysis of Human Data Set",
  Series = "GSA",
  M = 6783,
  Genome = "GRCh38",
  SourceType = "FASTQ",
  SourceUrl = "https://doi.org/10.5281/zenodo.1475409",
  SourceVersion = "md5:28508f1b03d6cfef75a7b537b7f186b6",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA_character_,
  DataProvider = "Genotype-Tissue Expression Project (GETx)",
  TestStatistic = "Enrichment Score",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)


## ##############################################################################
## GWAS case studies
## ##############################################################################

df_gwas_bmi_maf <- data.frame(
  Title = "bmi-maf-benchmark",
  Description = "Genome-Wide Association Analyses - BMI meta-analysis",
  Series = "GWAS",
  M = 196969,
  Genome = "hg18",
  SourceType = "Zip",
  SourceUrl = "http://portals.broadinstitute.org/collaboration/giant/images/3/3a/BMI.SNPadjSMK.zip",
  SourceVersion = "2017 GIANT GxSMK Project Files for Public Release",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = TRUE,
  DataProvider = "GIANT Consortium",
  TestStatistic = "t statistic",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Minor Allele Frequency",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_gwas_bmi_uninf <- data.frame(
  Title = "bmi-uninf-benchmark",
  Description = "Genome-Wide Association Analyses - BMI meta-analysis",
  Series = "GWAS",
  M = 196969,
  Genome = "hg18",
  SourceType = "Zip",
  SourceUrl = "http://portals.broadinstitute.org/collaboration/giant/images/3/3a/BMI.SNPadjSMK.zip",
  SourceVersion = "2017 GIANT GxSMK Project Files for Public Release",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = TRUE,
  DataProvider = "GIANT Consortium",
  TestStatistic = "t statistic",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_gwas_bmi_samplesize <- data.frame(
  Title = "bmi-samplesize-benchmark",
  Description = "Genome-Wide Association Analyses - BMI meta-analysis",
  Series = "GWAS",
  M = 196969,
  Genome = "hg18",
  SourceType = "Zip",
  SourceUrl = "http://portals.broadinstitute.org/collaboration/giant/images/3/3a/BMI.SNPadjSMK.zip",
  SourceVersion = "2017 GIANT GxSMK Project Files for Public Release",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = TRUE,
  DataProvider = "GIANT Consortium",
  TestStatistic = "t statistic",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Sample Size",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)


## ##############################################################################
## Microbiome analysis case studies
## ##############################################################################

df_baxter_otus <- data.frame(
  Title = "baxter-otus-benchmark",
  Description = "16S microbiome differential testing of Baxter et al. (2016) dataset at OTU-level with ubiquity as covariate",
  Series = "Microbiome",
  M = 9388,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/crc_baxter_results.tar.gz",
  SourceVersion = "md5:3604d25fe3ea0e48379d1f78a3aeb231",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "University of Michigan; SRA study SRP062005",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_baxter_otus_logubiquity <- data.frame(
  Title = "baxter-otus-log-ubiquity-benchmark",
  Description = "16S microbiome differential testing of Baxter et al. (2016) dataset at OTU-level with log ubiquity as covariate",
  Series = "Microbiome",
  M = 9388,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/crc_baxter_results.tar.gz",
  SourceVersion = "md5:3604d25fe3ea0e48379d1f78a3aeb231",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "University of Michigan; SRA study SRP062005",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Log ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_baxter_genus <- data.frame(
  Title = "baxter-genus-benchmark",
  Description = "16S microbiome differential testing of Baxter et al. (2016) dataset at genus-level with ubiquity as covariate",
  Series = "Microbiome",
  M = 148,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/crc_baxter_results.tar.gz",
  SourceVersion = "md5:3604d25fe3ea0e48379d1f78a3aeb231",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "University of Michigan; SRA study SRP062005",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_baxter_genus_logubiquity <- data.frame(
  Title = "baxter-genus-log-ubiquity-benchmark",
  Description = "16S microbiome differential testing of Baxter et al. (2016) dataset at genus-level with log ubiquity as covariate",
  Series = "Microbiome",
  M = 148,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/crc_baxter_results.tar.gz",
  SourceVersion = "md5:3604d25fe3ea0e48379d1f78a3aeb231",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "University of Michigan; SRA study SRP062005",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Log ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_baxter_genus_meanabun <- data.frame(
  Title = "baxter-genus-mean-abun-benchmark",
  Description = "16S microbiome differential testing of Baxter et al. (2016) dataset at genus-level with mean abundance as covariate",
  Series = "Microbiome",
  M = 148,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/crc_baxter_results.tar.gz",
  SourceVersion = "md5:3604d25fe3ea0e48379d1f78a3aeb231",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "University of Michigan; SRA study SRP062005",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Mean abundance",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_baxter_genus_uninf <- data.frame(
  Title = "baxter-genus-mean-uninf-benchmark",
  Description = "16S microbiome differential testing of Baxter et al. (2016) dataset at genus-level with uninformative covariate",
  Series = "Microbiome",
  M = 148,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/crc_baxter_results.tar.gz",
  SourceVersion = "md5:3604d25fe3ea0e48379d1f78a3aeb231",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "University of Michigan; SRA study SRP062005",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_goodrich_otus <- data.frame(
  Title = "goodrich-otus-benchmark",
  Description = "16S microbiome differential testing of Goodrich et al. (2014) dataset at OTU-level with ubiquity as covariate",
  Series = "Microbiome",
  M = 42388,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/ob_goodrich_results.tar.gz",
  SourceVersion = "md5:7aa4ee253d1c4ac51cbadd6ef366dea7",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "Colorado Center for Microbial Ecology, University of Colorado at Boulder; ENA studies PRJEB6702 and PRJEB6705",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_goodrich_genus <- data.frame(
  Title = "goodrich-genus-benchmark",
  Description = "16S microbiome differential testing of Goodrich et al. (2014) dataset at genus-level with ubiquity as covariate",
  Series = "Microbiome",
  M = 153,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/ob_goodrich_results.tar.gz",
  SourceVersion = "md5:7aa4ee253d1c4ac51cbadd6ef366dea7",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "Colorado Center for Microbial Ecology, University of Colorado at Boulder; ENA studies PRJEB6702 and PRJEB6705",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_goodrich_genus_abun <- data.frame(
  Title = "goodrich-genus-abun-benchmark",
  Description = "16S microbiome differential testing of Goodrich et al. (2014) dataset at genus-level with mean abundance as covariate",
  Series = "Microbiome",
  M = 153,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/ob_goodrich_results.tar.gz",
  SourceVersion = "md5:7aa4ee253d1c4ac51cbadd6ef366dea7",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "Colorado Center for Microbial Ecology, University of Colorado at Boulder; ENA studies PRJEB6702 and PRJEB6705",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Mean abundance",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_goodrich_genus_uninf <- data.frame(
  Title = "goodrich-genus-uninf-benchmark",
  Description = "16S microbiome differential testing of Goodrich et al. (2014) dataset at genus-level with uninformative covariate",
  Series = "Microbiome",
  M = 153,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/ob_goodrich_results.tar.gz",
  SourceVersion = "md5:7aa4ee253d1c4ac51cbadd6ef366dea7",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "Colorado Center for Microbial Ecology, University of Colorado at Boulder; ENA studies PRJEB6702 and PRJEB6705",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_papa_otus <- data.frame(
  Title = "papa-otus-benchmark",
  Description = "16S microbiome differential testing of Papa et al. (2012) dataset at OTU-level with ubiquity as covariate",
  Series = "Microbiome",
  M = 2355,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/ibd_alm_results.tar.gz",
  SourceVersion = "md5:e2d6037596ba692cb971e50a84f64290",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "Eric J. Alm, Massachusetts Institute of Technology, Cambridge, Massachusetts, USA <ejalm@mit.edu>",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_papa_genus <- data.frame(
  Title = "papa-genus-benchmark",
  Description = "16S microbiome differential testing of Papa et al. (2012) dataset at genus-level with ubiquity as covariate",
  Series = "Microbiome",
  M = 75,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/ibd_alm_results.tar.gz",
  SourceVersion = "md5:e2d6037596ba692cb971e50a84f64290",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "Eric J. Alm, Massachusetts Institute of Technology, Cambridge, Massachusetts, USA <ejalm@mit.edu>",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_papa_genus_abun <- data.frame(
  Title = "papa-genus-abun-benchmark",
  Description = "16S microbiome differential testing of Papa et al. (2012) dataset at genus-level with mean abundance as covariate",
  Series = "Microbiome",
  M = 75,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/ibd_alm_results.tar.gz",
  SourceVersion = "md5:e2d6037596ba692cb971e50a84f64290",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "Eric J. Alm, Massachusetts Institute of Technology, Cambridge, Massachusetts, USA <ejalm@mit.edu>",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Mean abundance",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_papa_genus_logubiquity <- data.frame(
  Title = "papa-genus-log-ubiquity-benchmark",
  Description = "16S microbiome differential testing of Papa et al. (2012) dataset at genus-level with log ubiquity as covariate",
  Series = "Microbiome",
  M = 75,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/ibd_alm_results.tar.gz",
  SourceVersion = "md5:e2d6037596ba692cb971e50a84f64290",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "Eric J. Alm, Massachusetts Institute of Technology, Cambridge, Massachusetts, USA <ejalm@mit.edu>",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Log ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_papa_genus_uninf <- data.frame(
  Title = "papa-genus-uninf-benchmark",
  Description = "16S microbiome differential testing of Papa et al. (2012) dataset at genus-level with uninformative covariate",
  Series = "Microbiome",
  M = 75,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/ibd_alm_results.tar.gz",
  SourceVersion = "md5:e2d6037596ba692cb971e50a84f64290",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "Eric J. Alm, Massachusetts Institute of Technology, Cambridge, Massachusetts, USA <ejalm@mit.edu>",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_schubert_otus <- data.frame(
  Title = "schubert-otus-benchmark",
  Description = "16S microbiome differential testing of Schubert et al. (2014) dataset at OTU-level with ubiquity as covariate",
  Series = "Microbiome",
  M = 3118,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/cdi_schubert_results.tar.gz",
  SourceVersion = "md5:26a2589c448d8425766db6b86155694f",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "Patrick Schloss, University of Michigan, Ann Arbor, MI, USA <mothur.org/CDI_MicrobiomeModeling>",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_schubert_otus_abun <- data.frame(
  Title = "schubert-otus-abun-benchmark",
  Description = "16S microbiome differential testing of Schubert et al. (2014) dataset at OTU-level with mean abundance as covariate",
  Series = "Microbiome",
  M = 3118,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/cdi_schubert_results.tar.gz",
  SourceVersion = "md5:26a2589c448d8425766db6b86155694f",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "Patrick Schloss, University of Michigan, Ann Arbor, MI, USA <mothur.org/CDI_MicrobiomeModeling>",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Mean abundance",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_schubert_otus_uninf <- data.frame(
  Title = "schubert-otus-uninf-benchmark",
  Description = "16S microbiome differential testing of Schubert et al. (2014) dataset at OTU-level with uninformative covariate",
  Series = "Microbiome",
  M = 3118,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://zenodo.org/record/840333/files/cdi_schubert_results.tar.gz",
  SourceVersion = "md5:26a2589c448d8425766db6b86155694f",
  Species = "Human gut metagenome",
  TaxonomyId = "408170",
  Coordinate_1_based = NA_character_,
  DataProvider = "Patrick Schloss, University of Michigan, Ann Arbor, MI, USA <mothur.org/CDI_MicrobiomeModeling>",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_enigma_al_otus <- data.frame(
  Title = "enigma-al-otus-benchmark",
  Description = "16S microbiome Spearman correlation test of ENIGMA dataset at OTU-level with Al measurements using ubiquity as covariate",
  Series = "Microbiome",
  M = 4007,
  Genome = NA_character_,
  SourceType = "TXT",
  SourceUrl = paste("https://zenodo.org/record/1455793/files/enigma.metadata.txt",
                    "https://zenodo.org/record/1455793/files/enigma.otu_table_resampled_updated_r.txt.gz",
                    collapse = ","),
  SourceVersion = paste("md5:210e7805e1f4ff000c0b2f1047b24f3c",
                        "md5:26927389fcce3be546f66c1031044744",
                        collapse = ","),
  Species = "groundwater metagenome",
  TaxonomyId = "717931",
  Coordinate_1_based = NA_character_,
  DataProvider = "ENIGMA consortium, John-Marc Chandonia, Physical Biosciences Division, Berkeley National Lab <https://www.mg-rast.org/linkin.cgi?project=mgp8190>",
  TestStatistic = "Spearman correlation",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_enigma_al_otus_uninf <- data.frame(
  Title = "enigma-al-otus-uninf-benchmark",
  Description = "16S microbiome Spearman correlation test of ENIGMA dataset at OTU-level with Al measurements using uninformative covariate",
  Series = "Microbiome",
  M = 4007,
  Genome = NA_character_,
  SourceType = "TXT",
  SourceUrl = paste("https://zenodo.org/record/1455793/files/enigma.metadata.txt",
                    "https://zenodo.org/record/1455793/files/enigma.otu_table_resampled_updated_r.txt.gz",
                    collapse = ","),
  SourceVersion = paste("md5:210e7805e1f4ff000c0b2f1047b24f3c",
                        "md5:26927389fcce3be546f66c1031044744",
                        collapse = ","),
  Species = "groundwater metagenome",
  TaxonomyId = "717931",
  Coordinate_1_based = NA_character_,
  DataProvider = "ENIGMA consortium, John-Marc Chandonia, Physical Biosciences Division, Berkeley National Lab <https://www.mg-rast.org/linkin.cgi?project=mgp8190>",
  TestStatistic = "Spearman correlation",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_enigma_ph_otus <- data.frame(
  Title = "enigma-ph-otus-benchmark",
  Description = "16S microbiome Spearman correlation test of ENIGMA dataset at OTU-level with pH measurements using ubiquity as covariate",
  Series = "Microbiome",
  M = 4007,
  Genome = NA_character_,
  SourceType = "TXT",
  SourceUrl = paste("https://zenodo.org/record/1455793/files/enigma.metadata.txt",
                    "https://zenodo.org/record/1455793/files/enigma.otu_table_resampled_updated_r.txt.gz",
                    collapse = ","),
  SourceVersion = paste("md5:210e7805e1f4ff000c0b2f1047b24f3c",
                        "md5:26927389fcce3be546f66c1031044744",
                        collapse = ","),
  Species = "groundwater metagenome",
  TaxonomyId = "717931",
  Coordinate_1_based = NA_character_,
  DataProvider = "ENIGMA consortium, John-Marc Chandonia, Physical Biosciences Division, Berkeley National Lab <https://www.mg-rast.org/linkin.cgi?project=mgp8190>",
  TestStatistic = "Spearman correlation",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_enigma_ph_otus_abun <- data.frame(
  Title = "enigma-ph-otus-abun-benchmark",
  Description = "16S microbiome Spearman correlation test of ENIGMA dataset at OTU-level with pH measurements using mean abundance as covariate",
  Series = "Microbiome",
  M = 4007,
  Genome = NA_character_,
  SourceType = "TXT",
  SourceUrl = paste("https://zenodo.org/record/1455793/files/enigma.metadata.txt",
                    "https://zenodo.org/record/1455793/files/enigma.otu_table_resampled_updated_r.txt.gz",
                    collapse = ","),
  SourceVersion = paste("md5:210e7805e1f4ff000c0b2f1047b24f3c",
                        "md5:26927389fcce3be546f66c1031044744",
                        collapse = ","),
  Species = "groundwater metagenome",
  TaxonomyId = "717931",
  Coordinate_1_based = NA_character_,
  DataProvider = "ENIGMA consortium, John-Marc Chandonia, Physical Biosciences Division, Berkeley National Lab <https://www.mg-rast.org/linkin.cgi?project=mgp8190>",
  TestStatistic = "Spearman correlation",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Mean abundance",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_enigma_ph_otus_uninf <- data.frame(
  Title = "enigma-ph-otus-uninf-benchmark",
  Description = "16S microbiome Spearman correlation test of ENIGMA dataset at OTU-level with pH measurements using uninformative covariate",
  Series = "Microbiome",
  M = 4007,
  Genome = NA_character_,
  SourceType = "TXT",
  SourceUrl = paste("https://zenodo.org/record/1455793/files/enigma.metadata.txt",
                    "https://zenodo.org/record/1455793/files/enigma.otu_table_resampled_updated_r.txt.gz",
                    collapse = ","),
  SourceVersion = paste("md5:210e7805e1f4ff000c0b2f1047b24f3c",
                        "md5:26927389fcce3be546f66c1031044744",
                        collapse = ","),
  Species = "groundwater metagenome",
  TaxonomyId = "717931",
  Coordinate_1_based = NA_character_,
  DataProvider = "ENIGMA consortium, John-Marc Chandonia, Physical Biosciences Division, Berkeley National Lab <https://www.mg-rast.org/linkin.cgi?project=mgp8190>",
  TestStatistic = "Spearman correlation",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)



df_enigma_so4_otus <- data.frame(
  Title = "enigma-so4-otus-benchmark",
  Description = "16S microbiome Spearman correlation test of ENIGMA dataset at OTU-level with SO4 measurements using ubiquity as covariate",
  Series = "Microbiome",
  M = 4007,
  Genome = NA_character_,
  SourceType = "TXT",
  SourceUrl = paste("https://zenodo.org/record/1455793/files/enigma.metadata.txt",
                    "https://zenodo.org/record/1455793/files/enigma.otu_table_resampled_updated_r.txt.gz",
                    collapse = ","),
  SourceVersion = paste("md5:210e7805e1f4ff000c0b2f1047b24f3c",
                        "md5:26927389fcce3be546f66c1031044744",
                        collapse = ","),
  Species = "groundwater metagenome",
  TaxonomyId = "717931",
  Coordinate_1_based = NA_character_,
  DataProvider = "ENIGMA consortium, John-Marc Chandonia, Physical Biosciences Division, Berkeley National Lab <https://www.mg-rast.org/linkin.cgi?project=mgp8190>",
  TestStatistic = "Spearman correlation",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Ubiquity",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_enigma_so4_otus_abun <- data.frame(
  Title = "enigma-so4-otus-abun-benchmark",
  Description = "16S microbiome Spearman correlation test of ENIGMA dataset at OTU-level with SO4 measurements using mean abundance as covariate",
  Series = "Microbiome",
  M = 4007,
  Genome = NA_character_,
  SourceType = "TXT",
  SourceUrl = paste("https://zenodo.org/record/1455793/files/enigma.metadata.txt",
                    "https://zenodo.org/record/1455793/files/enigma.otu_table_resampled_updated_r.txt.gz",
                    collapse = ","),
  SourceVersion = paste("md5:210e7805e1f4ff000c0b2f1047b24f3c",
                        "md5:26927389fcce3be546f66c1031044744",
                        collapse = ","),
  Species = "groundwater metagenome",
  TaxonomyId = "717931",
  Coordinate_1_based = NA_character_,
  DataProvider = "ENIGMA consortium, John-Marc Chandonia, Physical Biosciences Division, Berkeley National Lab <https://www.mg-rast.org/linkin.cgi?project=mgp8190>",
  TestStatistic = "Spearman correlation",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Mean abundance",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_enigma_so4_otus_uninf <- data.frame(
  Title = "enigma-so4-otus-uninf-benchmark",
  Description = "16S microbiome Spearman correlation test of ENIGMA dataset at OTU-level with SO4 measurements using uninformative covariate",
  Series = "Microbiome",
  M = 4007,
  Genome = NA_character_,
  SourceType = "TXT",
  SourceUrl = paste("https://zenodo.org/record/1455793/files/enigma.metadata.txt",
                    "https://zenodo.org/record/1455793/files/enigma.otu_table_resampled_updated_r.txt.gz",
                    collapse = ","),
  SourceVersion = paste("md5:210e7805e1f4ff000c0b2f1047b24f3c",
                        "md5:26927389fcce3be546f66c1031044744",
                        collapse = ","),
  Species = "groundwater metagenome",
  TaxonomyId = "717931",
  Coordinate_1_based = NA_character_,
  DataProvider = "ENIGMA consortium, John-Marc Chandonia, Physical Biosciences Division, Berkeley National Lab <https://www.mg-rast.org/linkin.cgi?project=mgp8190>",
  TestStatistic = "Spearman correlation",
  EffectSize = NA_character_,
  NullProportion = NA_real_,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

## ##############################################################################
## RNA-seq differential gene expression (DGE) case studies
## ##############################################################################

df_rnaseq_brain <- data.frame(
  Title = "brain-benchmark",
  Description = "Case study: Differential gene expression between two brain regions (nucleus accumbens and putamen) using data from the GTEx project.",
  Series = "RNA-seq",
  M = 0,
  Genome = "GRCh38",
  SourceType = "rds",
  SourceUrl = "https://zenodo.org/record/1475409/files/rnaseq-brain-counts.rds",
  SourceVersion = "1",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "Zenodo",
  TestStatistic = NA_character_,
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Mean of counts",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_rnaseq_brain_uninf <- data.frame(
  Title = "brain-uninf-benchmark",
  Description = "Case study: Differential gene expression between two brain regions (nucleus accumbens and putamen) using data from the GTEx project.",
  Series = "RNA-seq",
  M = 0,
  Genome = "GRCh38",
  SourceType = "rds",
  SourceUrl = "https://zenodo.org/record/1475409/files/rnaseq-brain-counts.rds",
  SourceVersion = "1",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "Zenodo",
  TestStatistic = NA_character_,
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Uninformative covariate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_rnaseq_mir200c <- data.frame(
  Title = "mir200c-benchmark",
  Description = "Case study: Differential expression using RNA-seq between mir200c knockdown cells and untreated cells.",
  Series = "RNA-seq",
  M = 0,
  Genome = "GRCh38",
  SourceType = "Rdata",
  SourceUrl = "http://duffel.rail.bio/recount/v2/SRP030475/rse_gene.Rdata",
  SourceVersion = "2",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = NA_character_,
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Mean of counts",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_rnaseq_mir200c_uninf <- data.frame(
  Title = "mir200c-uninf-benchmark",
  Description = "Case study: Differential expression using RNA-seq between mir200c knockdown cells and untreated cells.",
  Series = "RNA-seq",
  M = 0,
  Genome = "GRCh38",
  SourceType = "Rdata",
  SourceUrl = "http://duffel.rail.bio/recount/v2/SRP030475/rse_gene.Rdata",
  SourceVersion = "2",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = NA_character_,
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Uninformative covariate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)


## ##############################################################################
## Polyester in silico RNA-seq DGE case studies
## ##############################################################################

df_polyester_de5_uninfCov <- data.frame(
  Title = "polyester-results-de5-uninfCov",
  Description = "Polyester RNA-seq simulation study",
  Series = "PolyesterInSilico",
  M = 0,
  Genome = "",
  SourceType = "",
  SourceUrl = "",
  SourceVersion = "",
  Species = "",
  TaxonomyId = "",
  Coordinate_1_based = "",
  DataProvider = "",
  TestStatistic = "",
  EffectSize = "",
  NullProportion = "",
  InformativeCovariate = "",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_polyester_de5 <- data.frame(
  Title = "polyester-results-de5",
  Description = "Polyester RNA-seq simulation study",
  Series = "PolyesterInSilico",
  M = 0,
  Genome = "",
  SourceType = "",
  SourceUrl = "",
  SourceVersion = "",
  Species = "",
  TaxonomyId = "",
  Coordinate_1_based = "",
  DataProvider = "",
  TestStatistic = "",
  EffectSize = "",
  NullProportion = "",
  InformativeCovariate = "",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_polyester_de10_uninfCov <- data.frame(
  Title = "polyester-results-de10-uninfCov",
  Description = "Polyester RNA-seq simulation study",
  Series = "PolyesterInSilico",
  M = 0,
  Genome = "",
  SourceType = "",
  SourceUrl = "",
  SourceVersion = "",
  Species = "",
  TaxonomyId = "",
  Coordinate_1_based = "",
  DataProvider = "",
  TestStatistic = "",
  EffectSize = "",
  NullProportion = "",
  InformativeCovariate = "",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_polyester_de10 <- data.frame(
  Title = "polyester-results-de10",
  Description = "Polyester RNA-seq simulation study",
  Series = "PolyesterInSilico",
  M = 0,
  Genome = "",
  SourceType = "",
  SourceUrl = "",
  SourceVersion = "",
  Species = "",
  TaxonomyId = "",
  Coordinate_1_based = "",
  DataProvider = "",
  TestStatistic = "",
  EffectSize = "",
  NullProportion = "",
  InformativeCovariate = "",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_polyester_null5 <- data.frame(
  Title = "polyester-results-null5",
  Description = "Polyester RNA-seq simulation study",
  Series = "PolyesterInSilico",
  M = 0,
  Genome = "",
  SourceType = "",
  SourceUrl = "",
  SourceVersion = "",
  Species = "",
  TaxonomyId = "",
  Coordinate_1_based = "",
  DataProvider = "",
  TestStatistic = "",
  EffectSize = "",
  NullProportion = "",
  InformativeCovariate = "",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_polyester_null10 <- data.frame(
  Title = "polyester-results-null10",
  Description = "Polyester RNA-seq simulation study",
  Series = "PolyesterInSilico",
  M = 0,
  Genome = "",
  SourceType = "",
  SourceUrl = "",
  SourceVersion = "",
  Species = "",
  TaxonomyId = "",
  Coordinate_1_based = "",
  DataProvider = "",
  TestStatistic = "",
  EffectSize = "",
  NullProportion = "",
  InformativeCovariate = "",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_polyesterW_de5 <- data.frame(
  Title = "polyesterW-results-de5",
  Description = "Polyester RNA-seq simulation study",
  Series = "PolyesterInSilico",
  M = 0,
  Genome = "",
  SourceType = "",
  SourceUrl = "",
  SourceVersion = "",
  Species = "",
  TaxonomyId = "",
  Coordinate_1_based = "",
  DataProvider = "",
  TestStatistic = "",
  EffectSize = "",
  NullProportion = "",
  InformativeCovariate = "",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_polyesterW_de10 <- data.frame(
  Title = "polyesterW-results-de10",
  Description = "Polyester RNA-seq simulation study",
  Series = "PolyesterInSilico",
  M = 0,
  Genome = "",
  SourceType = "",
  SourceUrl = "",
  SourceVersion = "",
  Species = "",
  TaxonomyId = "",
  Coordinate_1_based = "",
  DataProvider = "",
  TestStatistic = "",
  EffectSize = "",
  NullProportion = "",
  InformativeCovariate = "",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)


## ##############################################################################
## Yeast in silico RNA-seq DGE case studies
## ##############################################################################

df_yeast_de5_uninfCov <- data.frame(
  Title = "yeast-results-de5-uninfCov",
  Description = "Yeast RNA-seq 48 sample simulation study: Unimodal alternative, 2000 DE genes, uninformative covariate, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 0.695,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeast_de5 <- data.frame(
  Title = "yeast-results-de5",
  Description = "Yeast RNA-seq 48 sample simulation study: Unimodal alternative, 2000 DE genes, strong informative covariate, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 0.695,
  InformativeCovariate = "Strong",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeast_de10_uninfCov <- data.frame(
  Title = "yeast-results-de10-uninfCov",
  Description = "Yeast RNA-seq 48 sample simulation study: Unimodal alternative, 2000 DE genes, uninformative covariate, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 0.695,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeast_de10 <- data.frame(
  Title = "yeast-results-de10",
  Description = "Yeast RNA-seq 48 sample simulation study: Unimodal alternative, 2000 DE genes, strong informative covariate, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 0.695,
  InformativeCovariate = "Strong",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeast_null5 <- data.frame(
  Title = "yeast-results-null5",
  Description = "Yeast RNA-seq 48 sample simulation study: Null, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 1,
  InformativeCovariate = "none",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeast_null10 <- data.frame(
  Title = "yeast-results-null10",
  Description = "Yeast RNA-seq 48 sample simulation study: Null, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 1,
  InformativeCovariate = "none",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastH_de5_uninfCov <- data.frame(
  Title = "yeastH-results-de5-uninfCov",
  Description = "Yeast RNA-seq 48 sample simulation study: Unimodal alternative, 500 DE genes, uninformative covariate, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 0.924,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastH_de5 <- data.frame(
  Title = "yeastH-results-de5",
  Description = "Yeast RNA-seq 48 sample simulation study: Unimodal alternative, 500 DE genes, strong informative covariate, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 0.924,
  InformativeCovariate = "Strong",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastH_de10_uninfCov <- data.frame(
  Title = "yeastH-results-de10-uninfCov",
  Description = "Yeast RNA-seq 48 sample simulation study: Unimodal alternative, 500 DE genes, uninformative covariate, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 0.924,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastH_de10 <- data.frame(
  Title = "yeastH-results-de10",
  Description = "Yeast RNA-seq 48 sample simulation study: Unimodal alternative, 500 DE genes, strong informative covariate, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 0.924,
  InformativeCovariate = "Strong",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastHW_de5 <- data.frame(
  Title = "yeastHW-results-de5",
  Description = "Yeast RNA-seq 48 sample simulation study: Unimodal alternative, 500 DE genes, weak informative covariate, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 0.924,
  InformativeCovariate = "Weak",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastHW_de10 <- data.frame(
  Title = "yeastHW-results-de10",
  Description = "Yeast RNA-seq 48 sample simulation study: Unimodal alternative, 500 DE genes, weak informative covariate, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 0.924,
  InformativeCovariate = "Weak",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastII_de5_uninfCov <- data.frame(
  Title = "yeastII-results-de5-uninfCov",
  Description = "Yeast RNA-seq 48 sample simulation study: Bimodal alternative, 2000 DE genes, uninformative covariate, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Bimodal",
  NullProportion = 0.695,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastII_de5 <- data.frame(
  Title = "yeastII-results-de5",
  Description = "Yeast RNA-seq 48 sample simulation study: Bimodal alternative, 2000 DE genes, strong informative covariate, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Bimodal",
  NullProportion = 0.695,
  InformativeCovariate = "Strong",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastII_de10_uninfCov <- data.frame(
  Title = "yeastII-results-de10-uninfCov",
  Description = "Yeast RNA-seq 48 sample simulation study: Bimodal alternative, 2000 DE genes, uninformative covariate, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Bimodal",
  NullProportion = 0.695,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastII_de10 <- data.frame(
  Title = "yeastII-results-de10",
  Description = "Yeast RNA-seq 48 sample simulation study: Bimodal alternative, 2000 DE genes, strong informative covariate, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Bimodal",
  NullProportion = 0.695,
  InformativeCovariate = "Strong",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastIIH_de5_uninfCov <- data.frame(
  Title = "yeastIIH-results-de5-uninfCov",
  Description = "Yeast RNA-seq 48 sample simulation study: bimodal alternative, 500 DE genes, uninformative covariate, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Bimodal",
  NullProportion = 0.924,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastIIH_de5 <- data.frame(
  Title = "yeastIIH-results-de5",
  Description = "Yeast RNA-seq 48 sample simulation study: bimodal alternative, 500 DE genes, strong informative covariate, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Bimodal",
  NullProportion = 0.924,
  InformativeCovariate = "Strong",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastIIH_de10_uninfCov <- data.frame(
  Title = "yeastIIH-results-de10-uninfCov",
  Description = "Yeast RNA-seq 48 sample simulation study: bimodal alternative, 500 DE genes, uninformative covariate, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Bimodal",
  NullProportion = 0.924,
  InformativeCovariate = "Uninformative",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastIIH_de10 <- data.frame(
  Title = "yeastIIH-results-de10",
  Description = "Yeast RNA-seq 48 sample simulation study: bimodal alternative, 500 DE genes, strong informative covariate, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Bimodal",
  NullProportion = 0.924,
  InformativeCovariate = "Strong",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastIIHW_de5 <- data.frame(
  Title = "yeastIIHW-results-de5",
  Description = "Yeast RNA-seq 48 sample simulation study: bimodal alternative, 500 DE genes, weak informative covariate, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Bimodal",
  NullProportion = 0.924,
  InformativeCovariate = "Weak",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastIIHW_de10 <- data.frame(
  Title = "yeastIIHW-results-de10",
  Description = "Yeast RNA-seq 48 sample simulation study: bimodal alternative, 500 DE genes, weak informative covariate, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Bimodal",
  NullProportion = 0.924,
  InformativeCovariate = "Weak",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastIIW_de5 <- data.frame(
  Title = "yeastIIW-results-de5",
  Description = "Yeast RNA-seq 48 sample simulation study: bimodal alternative, 2000 DE genes, weak informative covariate, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Bimodal",
  NullProportion = 0.695,
  InformativeCovariate = "Weak",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastIIW_de10 <- data.frame(
  Title = "yeastIIW-results-de10",
  Description = "Yeast RNA-seq 48 sample simulation study: bimodal alternative, 2000 DE genes, weak informative covariate, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Bimodal",
  NullProportion = 0.695,
  InformativeCovariate = "Weak",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastW_de5 <- data.frame(
  Title = "yeastW-results-de5",
  Description = "Yeast RNA-seq 48 sample simulation study: Unimodal alternative, 2000 DE genes, weak informative covariate, 5 vs. 5 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 0.695,
  InformativeCovariate = "Weak",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_yeastW_de10 <- data.frame(
  Title = "yeastW-results-de10",
  Description = "Yeast RNA-seq 48 sample simulation study: Unimodal alternative, 2000 DE genes, weak informative covariate, 10 vs. 10 comparison",
  Series = "YeastInSilico",
  M = 6553,
  Genome = NA_character_,
  SourceType = "tar.gz",
  SourceUrl = "https://github.com/bartongroup/profDGE48",
  SourceVersion = "375dc0d57d9d1fa96a4245a6530e0fda34305891",
  Species = "Saccharomyces cerevisiae",
  TaxonomyId = NA_character_,
  Coordinate_1_based = NA,
  DataProvider = "",
  TestStatistic = "DESeq2",
  EffectSize = "Unimodal",
  NullProportion = 0.695,
  InformativeCovariate = "Weak",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)


## ##############################################################################
## single-cell RNA-seq DGE case studies
## ##############################################################################

df_scrnaseq_human_mast_det <- data.frame(
  Title = "human-benchmark-mast-det",
  Description = "Case study: differential expression using the method MAST on single-cell RNAseq data of diffuse neoplastic infiltrating cells at the migrating front of human glioblastoma.",
  Series = "scRNA-seq",
  M = 23257,
  Genome = "GRCh38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE84465.rds",
  SourceVersion = "1",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Chi-squared",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Detection rate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_human_mast_mean <- data.frame(
  Title = "human-benchmark-mast-mean",
  Description = "Case study: differential expression using the method MAST on single-cell RNAseq data of diffuse neoplastic infiltrating cells at the migrating front of human glioblastoma.",
  Series = "scRNA-seq",
  M = 23257,
  Genome = "GRCh38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE84465.rds",
  SourceVersion = "1",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Chi-squared",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Mean of counts",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_human_mast_uninf <- data.frame(
  Title = "human-benchmark-mast-uninf",
  Description = "Case study: differential expression using the method MAST on single-cell RNAseq data of diffuse neoplastic infiltrating cells at the migrating front of human glioblastoma.",
  Series = "scRNA-seq",
  M = 23257,
  Genome = "GRCh38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE84465.rds",
  SourceVersion = "1",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Chi-squared",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Uninformative covariate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_human_scdd_det <- data.frame(
  Title = "human-benchmark-scdd-det",
  Description = "Case study: differential expression using the method scDD on single-cell RNAseq data of diffuse neoplastic infiltrating cells at the migrating front of human glioblastoma.",
  Series = "scRNA-seq",
  M = 23257,
  Genome = "GRCh38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE84465.rds",
  SourceVersion = "1",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Permutation Test",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Detection rate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_human_scdd_mean <- data.frame(
  Title = "human-benchmark-scdd-mean",
  Description = "Case study: differential expression using the method scDD on single-cell RNAseq data of diffuse neoplastic infiltrating cells at the migrating front of human glioblastoma.",
  Series = "scRNA-seq",
  M = 23257,
  Genome = "GRCh38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE84465.rds",
  SourceVersion = "1",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Permutation Test",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Mean of counts",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_human_scdd_uninf <- data.frame(
  Title = "human-benchmark-scdd-uninf",
  Description = "Case study: differential expression using the method scDD on single-cell RNAseq data of diffuse neoplastic infiltrating cells at the migrating front of human glioblastoma.",
  Series = "scRNA-seq",
  M = 23257,
  Genome = "GRCh38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE84465.rds",
  SourceVersion = "1",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Permutation Test",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Uninformative covariate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_human_wilcox_det <- data.frame(
  Title = "human-benchmark-wilcox-det",
  Description = "Case study: differential expression using a Wilcox test on single-cell RNAseq data of diffuse neoplastic infiltrating cells at the migrating front of human glioblastoma.",
  Series = "scRNA-seq",
  M = 23257,
  Genome = "GRCh38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE84465.rds",
  SourceVersion = "1",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Detection rate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_human_wilcox_mean <- data.frame(
  Title = "human-benchmark-wilcox-mean",
  Description = "Case study: differential expression using a Wilcox test on single-cell RNAseq data of diffuse neoplastic infiltrating cells at the migrating front of human glioblastoma.",
  Series = "scRNA-seq",
  M = 23257,
  Genome = "GRCh38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE84465.rds",
  SourceVersion = "1",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Mean of counts",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_human_wilcox_uninf <- data.frame(
  Title = "human-benchmark-wilcox-uninf",
  Description = "Case study: differential expression using a Wilcox test on single-cell RNAseq data of diffuse neoplastic infiltrating cells at the migrating front of human glioblastoma.",
  Series = "scRNA-seq",
  M = 23257,
  Genome = "GRCh38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE84465.rds",
  SourceVersion = "1",
  Species = "Homo sapiens",
  TaxonomyId = "9606",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Uninformative covariate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_mouse_mast_det <- data.frame(
  Title = "mouse-benchmark-mast-det",
  Description = "Case study: differential expression using the method MAST on single mouse cells upon NF-kappa-beta activation.",
  Series = "scRNA-seq",
  M = 13777,
  Genome = "GRCm38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE94383.rds",
  SourceVersion = "1",
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Chi-squared",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Detection rate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_mouse_mast_mean <- data.frame(
  Title = "mouse-benchmark-mast-mean",
  Description = "Case study: differential expression using the method MAST on single mouse cells upon NF-kappa-beta activation.",
  Series = "scRNA-seq",
  M = 13777,
  Genome = "GRCm38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE94383.rds",
  SourceVersion = "1",
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Chi-squared",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Mean of counts",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_mouse_mast_uninf <- data.frame(
  Title = "mouse-benchmark-mast-uninf",
  Description = "Case study: differential expression using the method MAST on single mouse cells upon NF-kappa-beta activation.",
  Series = "scRNA-seq",
  M = 13777,
  Genome = "GRCm38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE94383.rds",
  SourceVersion = "1",
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Chi-squared",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Uninformative covariate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_mouse_scdd_det <- data.frame(
  Title = "mouse-benchmark-scdd-det",
  Description = "Case study: differential expression using the method scDD on single mouse cells upon NF-kappa-beta activation.",
  Series = "scRNA-seq",
  M = 13777,
  Genome = "GRCm38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE94383.rds",
  SourceVersion = "1",
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Permutation Test",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Detection rate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_mouse_scdd_mean <- data.frame(
  Title = "mouse-benchmark-scdd-mean",
  Description = "Case study: differential expression using the method scDD on single mouse cells upon NF-kappa-beta activation.",
  Series = "scRNA-seq",
  M = 13777,
  Genome = "GRCm38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE94383.rds",
  SourceVersion = "1",
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Permutation Test",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Mean of counts",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_mouse_scdd_uninf <- data.frame(
  Title = "mouse-benchmark-scdd-uninf",
  Description = "Case study: differential expression using the method scDD on single mouse cells upon NF-kappa-beta activation.",
  Series = "scRNA-seq",
  M = 13777,
  Genome = "GRCm38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE94383.rds",
  SourceVersion = "1",
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Permutation Test",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Uninformative covariate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_mouse_wilcox_det <- data.frame(
  Title = "mouse-benchmark-wilcox-det",
  Description = "Case study: differential expression using a Wilcox test on single mouse cells upon NF-kappa-beta activation.",
  Series = "scRNA-seq",
  M = 13777,
  Genome = "GRCm38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE94383.rds",
  SourceVersion = "1",
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Detection rate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_mouse_wilcox_mean <- data.frame(
  Title = "mouse-benchmark-wilcox-mean",
  Description = "Case study: differential expression using a Wilcox test on single mouse cells upon NF-kappa-beta activation.",
  Series = "scRNA-seq",
  M = 13777,
  Genome = "GRCm38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE94383.rds",
  SourceVersion = "1",
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Mean of counts",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

df_scrnaseq_mouse_wilcox_uninf <- data.frame(
  Title = "mouse-benchmark-wilcox-uninf",
  Description = "Case study: differential expression using a Wilcox test on single mouse cells upon NF-kappa-beta activation.",
  Series = "scRNA-seq",
  M = 13777,
  Genome = "GRCm38",
  SourceType = "rds",
  SourceUrl = "http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/GSE94383.rds",
  SourceVersion = "1",
  Species = "Mus musculus",
  TaxonomyId = "10090",
  Coordinate_1_based = NA,
  DataProvider = "Conquer database",
  TestStatistic = "Wilcoxon rank-sum",
  EffectSize = NA_character_,
  NullProportion = NA_character_,
  InformativeCovariate = "Uninformative covariate",
  RDataClass = "SummarizedBenchmark",
  DispatchClass = "Rds",
  BiocVersion = "3.10",
  Maintainer = "Keegan Korthauer <kdkorthauer@gmail.com>",
  stringsAsFactors = FALSE
)

## ##############################################################################
## ##############################################################################

## Combined meta-data
df_all_casestudies <- base::rbind(
  df_h3k4me3_promoters,
  df_h3k4me3_csaw,
  df_h3k4me3_csaw_uninf,
  df_h3k4me3_csaw_cov,
  df_cbp_csaw,
  df_cbp_csaw_uninf,
  df_cbp_csaw_cov,
  df_gsa_mouse,
  df_gsa_mouse_uninf,
  df_gsa_human,
  df_gsa_human_uninf,
  df_gwas_bmi_maf,
  df_gwas_bmi_uninf,
  df_gwas_bmi_samplesize,
  df_baxter_genus,
  df_baxter_genus_logubiquity,
  df_baxter_genus_meanabun,
  df_baxter_genus_uninf,
  df_baxter_otus,
  df_baxter_otus_logubiquity,
  df_enigma_al_otus,
  df_enigma_al_otus_uninf,
  df_enigma_ph_otus,
  df_enigma_ph_otus_abun,
  df_enigma_ph_otus_uninf,
  df_enigma_so4_otus,
  df_enigma_so4_otus_abun,
  df_enigma_so4_otus_uninf,
  df_goodrich_genus,
  df_goodrich_genus_abun,
  df_goodrich_genus_uninf,
  df_goodrich_otus,
  df_papa_genus,
  df_papa_genus_abun,
  df_papa_genus_logubiquity,
  df_papa_genus_uninf,
  df_papa_otus,
  df_schubert_otus,
  df_schubert_otus_abun,
  df_schubert_otus_uninf,
  df_rnaseq_brain,
  df_rnaseq_brain_uninf,
  df_rnaseq_mir200c,
  df_rnaseq_mir200c_uninf,
  df_polyester_de5,
  df_polyester_de5_uninfCov,
  df_polyester_de10,
  df_polyester_de10_uninfCov,
  df_polyester_null5,
  df_polyester_null10,
  df_polyesterW_de5,
  df_polyesterW_de10,
  df_yeast_de5,
  df_yeast_de5_uninfCov,
  df_yeast_de10,
  df_yeast_de10_uninfCov,
  df_yeast_null5,
  df_yeast_null10,
  df_yeastH_de5,
  df_yeastH_de5_uninfCov,
  df_yeastH_de10,
  df_yeastH_de10_uninfCov,
  df_yeastHW_de5,
  df_yeastHW_de10,
  df_yeastII_de5,
  df_yeastII_de5_uninfCov,
  df_yeastII_de10,
  df_yeastII_de10_uninfCov,
  df_yeastIIH_de5,
  df_yeastIIH_de5_uninfCov,
  df_yeastIIH_de10,
  df_yeastIIH_de10_uninfCov,
  df_yeastIIHW_de5,
  df_yeastIIHW_de10,
  df_yeastIIW_de5,
  df_yeastIIW_de10,
  df_yeastW_de5,
  df_yeastW_de10,
  df_scrnaseq_human_mast_det,
  df_scrnaseq_human_mast_mean,
  df_scrnaseq_human_mast_uninf,
  df_scrnaseq_human_scdd_det,
  df_scrnaseq_human_scdd_mean,
  df_scrnaseq_human_scdd_uninf,
  df_scrnaseq_human_wilcox_det,
  df_scrnaseq_human_wilcox_mean,
  df_scrnaseq_human_wilcox_uninf,
  df_scrnaseq_mouse_mast_det,
  df_scrnaseq_mouse_mast_mean,
  df_scrnaseq_mouse_mast_uninf,
  df_scrnaseq_mouse_scdd_det,
  df_scrnaseq_mouse_scdd_mean,
  df_scrnaseq_mouse_scdd_uninf,
  df_scrnaseq_mouse_wilcox_det,
  df_scrnaseq_mouse_wilcox_mean,
  df_scrnaseq_mouse_wilcox_uninf
)


## ##############################################################################
## ##############################################################################

## all results saved as RDS in simulations subfolder w/ Title matching file name
df_all_casestudies$RDataPath <-
    paste0("benchmarkfdrData2019/", df_all_casestudies$Series, "/", df_all_casestudies$Title, ".rds")


## ##############################################################################
## ##############################################################################

write.csv(file = "inst/extdata/metadata-casestudies.csv",
          df_all_casestudies, row.names = FALSE)

